plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -417,30 +419,36 @@ public boolean accept(File pathname) {
     }
 
     private KubernetesListBuilder generateAppResources(List<ImageConfiguration> images, EnricherManager enricherManager) throws IOException, MojoExecutionException {
-        File[] resourceFiles = KubernetesResourceUtil.listResourceFragments(resourceDir);
-        KubernetesListBuilder builder;
+        try {
+            File[] resourceFiles = KubernetesResourceUtil.listResourceFragments(resourceDir);
+            KubernetesListBuilder builder;
 
-        // Add resource files found in the fabric8 directory
-        if (resourceFiles != null && resourceFiles.length > 0) {
-            log.info("Using resource templates from %s", resourceDir);
-            builder = readResourceFragments(resourceFiles);
-        } else {
-            builder = new KubernetesListBuilder();
-        }
+            // Add resource files found in the fabric8 directory
+            if (resourceFiles != null && resourceFiles.length > 0) {
+                log.info("Using resource templates from %s", resourceDir);
+                builder = readResourceFragments(resourceFiles);
+            } else {
+                builder = new KubernetesListBuilder();
+            }
 
-        // Add locally configured objects
-        if (resources != null) {
-            // TODO: Allow also support resources to be specified via XML
-            addConfiguredResources(builder, images);
-        }
+            // Add locally configured objects
+            if (resources != null) {
+                // TODO: Allow also support resources to be specified via XML
+                addConfiguredResources(builder, images);
+            }
 
-        // Create default resources for app resources only
-        enricherManager.createDefaultResources(builder);
+            // Create default resources for app resources only
+            enricherManager.createDefaultResources(builder);
 
-        // Enrich descriptors
-        enricherManager.enrich(builder);
+            // Enrich descriptors
+            enricherManager.enrich(builder);
 
-        return builder;
+            return builder;
+        } catch (ConstraintViolationException e) {
+            String message = ValidationUtil.createValidationMessage(e.getConstraintViolations());
+            getLog().error("ConstraintViolationException: " + message, e);
```

Comment:
when you throw a ConstraintViolationException you have a choice to pass in a meaningful text message that will appear in any output in, say, maven or JUnit. 

However kubernetes-model chooses not to pass in any message at all (and the base class of ConstraintViolationException helpfully chooses not to pass any default message either even if you have 20 violations :)

it'd be nice for all statements like `throw new ConstraintViolationException(violations)` in kubernetes model to use a helper method to generate a human useful error message. 

I've raised a separate issue to address this generically in kubernetes-model: https://github.com/fabric8io/kubernetes-model/issues/178